nan.h

```diff
@@ -836,9 +936,15 @@ class TryCatch {
     return scope.Escape(New(node::MakeCallback(
         v8::Isolate::GetCurrent(), target, func, argc, argv)));
 #else
+# if NODE_MODULE_VERSION >= NODE_9_0_MODULE_VERSION
+    AsyncResource res("nan:makeCallback");
+    return res.runInAsyncScope(target, func, argc, argv)
```

Comment:
I am afraid that this defeats the purpose behind the deprecation of certain `MakeCallbacks` in Node.js 10.

The idea is that the `AsyncResource` should be constructed when a callback is 'stored' for later invocation. That way the async context can be restored at the callback invocation time. Creating the resource here would not restore the correct async context – it would be a no-op.

It would indeed get rid of the deprecation warning in nan itself though.

We perhaps need to deprecate these particular `MakeCallback` APIs in Nan, so that consumer native modules get notified?

Reply:
IOW, let's silence the node warning, but give warning to `Nan` users if they use these functions by deprecating these `MakeCallback`s.